main.cpp

```diff
@@ -61,124 +69,335 @@ New/This/Pointers/References conclusion
+    FloatType& add ( const IntType& );
+    FloatType& subtract ( const IntType& );
+    FloatType& multiply ( const IntType& );
+    FloatType& divide ( const IntType& );
 };
 
-float FloatType::add (float x, float y)
+FloatType& FloatType::add ( const float x )
 {
-    return (x+y);
+    *pointerToFloatValue += x;
+    return *this;
 }
 
-float FloatType::subtract (float x, float y) 
+FloatType& FloatType::subtract ( const float y ) 
 { 
-    return (x-y); 
+    *pointerToFloatValue -= y;
+    return *this; 
 }
 
-float FloatType::multiply (float x, float y) 
+FloatType& FloatType::multiply ( const float y) 
 { 
-    return (x*y); 
+    *pointerToFloatValue *= y;
+    return *this; 
 }
 
-float FloatType::divide (float x, float y) 
+FloatType& FloatType::divide ( const float y ) 
 { 
-    if (abs(y) > 0) return (x/y);    
-    
-    std::cout << "Divide-by-zero warning  " << std::endl;
-    return (x/y);      
+    std::string warning = (abs(y) > 0) ? "" : "\x1B[31m Divide-by-zero warning! \x1B[0m";
+    std::cout << warning << std::endl;   
+
+    *pointerToFloatValue /= y; 
+    return *this;      
 }
 
 struct DoubleType
 {
-    double add (double, double);
-    double subtract ( double, double);
-    double multiply ( double, double);
-    double divide (double, double);
+    double* pointerToDoubleValue = nullptr;
+
+    DoubleType(double doubleIn )
+    {
+        pointerToDoubleValue = new double( doubleIn );
+    }
+    
+    ~DoubleType()
+    {
+        delete pointerToDoubleValue;
+        pointerToDoubleValue = nullptr;
+    }
+
+    DoubleType& add ( double );
+    DoubleType& subtract ( double );
+    DoubleType& multiply ( double );
+    DoubleType& divide ( double );
+    DoubleType& add ( const FloatType& );
+    DoubleType& subtract ( const FloatType& );
+    DoubleType& multiply ( const FloatType& );
+    DoubleType& divide ( const FloatType& );
+    DoubleType& add ( const IntType& );
+    DoubleType& subtract ( const IntType& );
+    DoubleType& multiply ( const IntType& );
+    DoubleType& divide ( const IntType& );
 };
 
-double DoubleType::add (double x, double y)
+DoubleType& DoubleType::add ( const double y)
 {
-    return (x+y);
+    *pointerToDoubleValue += y;
+    return *this;
 }
 
-double DoubleType::subtract (double x, double y) 
+DoubleType& DoubleType::subtract ( const double y) 
 { 
-    return (x-y); 
+    *pointerToDoubleValue -= y;
+    return *this; 
 }
 
-double DoubleType::multiply (double x, double y) 
+DoubleType& DoubleType::multiply ( const double y) 
 { 
-    return (x*y); 
+    *pointerToDoubleValue *= y;
+    return *this; 
 }
 
-double DoubleType::divide (double x, double y) 
+DoubleType& DoubleType::divide ( const double y) 
 { 
-    if (abs(y) > 0) return (x/y);    
-    
-    std::cout << "Divide-by-zero warning  " << std::endl;
-    return (x/y);      
+    std::string warning = (abs(y) > 0) ? "" : "\x1B[31m Divide-by-zero warning! \x1B[0m";
+    std::cout << warning << std::endl;
```

Comment:
why not simply:
```
if( abs(y) > 0 )
    std::cout << "warning: floating point division by zero\n";
```
You're constructing a string that is potentially empty, and then printing it regardless of whether or not it is an empty string or not.

Reply:
in your suggestion, a non-zero denominator would show the warning.